django/contrib/admin/filters.py

```diff
@@ -257,10 +257,14 @@ def field_admin_ordering(self, field, request, model_admin):
         """
         Return the model admin's ordering for related field, if provided.
         """
-        related_admin = model_admin.admin_site._registry.get(field.remote_field.model)
-        if related_admin is not None:
-            return related_admin.get_ordering(request)
-        return ()
+        from django.contrib.admin.sites import NotRegistered
+
+        try:
+            return model_admin.admin_site.get_model_admin(
+                field.remote_field.model
+            ).get_ordering(request)
+        except NotRegistered:
+            return ()
```

Comment:
Keep stuff in the try-block to a minimum?

```suggestion
        try:
            related_admin = model_admin.admin_site.get_model_admin(
                field.remote_field.model
            )
        except NotRegistered:
            return ()
        else:
            return related_admin.get_ordering(request)
```